src/main.rs

```diff
@@ -675,30 +538,139 @@ async fn main() {
+                    .await
+                    .context("finalized head")?;
+
+                let head_key = format!(
+                    "0x{}{}",
+                    hex::encode(twox_128(pallet.as_bytes())),
+                    hex::encode(twox_128(head.as_bytes()))
+                );
+                let head_str: String = state
+                    .avail_client
+                    .request(
+                        "state_getStorage",
+                        rpc_params![head_key, finalized_block_hash_str.clone()],
+                    )
+                    .await
+                    .context("head key")?;
+
+                let slot_from_hex =
+                    sp_core::bytes::from_hex(head_str.as_str()).context("decode slot")?;
+                let slot: u64 =
+                    Decode::decode(&mut slot_from_hex.as_slice()).context("slot decode 2")?;
+
+                let timestamp_key = format!(
+                    "0x{}{}{}",
+                    hex::encode(twox_128(pallet.as_bytes())),
+                    hex::encode(twox_128(timestamp.as_bytes())),
+                    &head_str[2..].to_string()
+                );
+
+                let timestamp_str: String = state
+                    .avail_client
+                    .request(
+                        "state_getStorage",
+                        rpc_params![timestamp_key, finalized_block_hash_str],
+                    )
+                    .await
+                    .context("timestamp key")?;
+
+                let timestamp_from_hex =
+                    sp_core::bytes::from_hex(timestamp_str.as_str()).context("timestamp decode")?;
+
+                let timestamp: u64 = Decode::decode(&mut timestamp_from_hex.as_slice())
+                    .context("timestamp decode 2")?;
+
+                let slot_block_head = SLOT_BLOCK_HEAD.lock().await;
+                if let Some((old_slot, _old_block, _old_hash, _old_timestamp)) =
+                    slot_block_head.as_ref()
+                {
+                    if old_slot == &slot {
+                        tokio::time::sleep(Duration::from_secs(60 * 10)).await;
+                        continue;
+                    }
+                }
+
+                drop(slot_block_head);
+
+                let resp = reqwest::get(format!("{}/{}", state.beaconchain_base_url, slot))
+                    .await
+                    .context("beacon get")?;
+                let res = resp
+                    .json::<BeaconAPIResponse>()
+                    .await
+                    .context("beacon decode")?;
+                let bl = res.data.exec_block_number;
+                let h = res.data.exec_block_hash;
+                let mut slot_block_head = SLOT_BLOCK_HEAD.lock().await;
+                tracing::info!("Beacon mapping: {slot}:{bl}");
+                *slot_block_head = Some((slot, bl as u64, h, timestamp));
+
+                tokio::time::sleep(Duration::from_secs(60 * 5)).await;
```

Comment:
Can we delegate this to a `tokio` / `tower` / `axum` cronjob?

Reply:
Is there a huge difference? What would be the benefit? Using system services? If so, I'll change it.

Reply:
It would not be very different _for now_, just for maintainability.

Reply:
I've decided to make a task for future for this, there are multiple solutions, none of which are really idiomatic/built-in in the already used crates, so I think we can defer it.